lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.Model.rb

```diff
@@ -657,31 +657,120 @@ def handle_user_input_data(model, climate_zone, default_hvac_building_type, defa
-              air_loop.thermalZones.each do |thermal_zone|
-                thermal_zone.additionalProperties.setFeature('economizer_exception_for_gas_phase_air_cleaning', air_loop_name)
+            if user_airloop.key?('economizer_exception_for_gas_phase_air_cleaning') && !user_airloop['economizer_exception_for_gas_phase_air_cleaning'].nil?
+              if user_airloop['economizer_exception_for_gas_phase_air_cleaning'].downcase == 'yes'
+                air_loop.thermalZones.each do |thermal_zone|
+                  thermal_zone.additionalProperties.setFeature('economizer_exception_for_gas_phase_air_cleaning', air_loop_name)
+                end
               end
             end
             # Open refrigerated cases is zone based - add yes or no to zones
-            if user_airloop.key?('economizer_exception_for_open_refrigerated_cases') &&
-               user_airloop['economizer_exception_for_open_refrigerated_cases'].downcase == 'yes'
-              air_loop.thermalZones.each do |thermal_zone|
-                thermal_zone.additionalProperties.setFeature('economizer_exception_for_open_refrigerated_cases', 'yes')
+            if user_airloop.key?('economizer_exception_for_open_refrigerated_cases') && !user_airloop['economizer_exception_for_open_refrigerated_cases'].nil?
+              if user_airloop['economizer_exception_for_open_refrigerated_cases'].downcase == 'yes'
+                air_loop.thermalZones.each do |thermal_zone|
+                  thermal_zone.additionalProperties.setFeature('economizer_exception_for_open_refrigerated_cases', 'yes')
+                end
+              end
+            end
+            # Fan power credits
+            user_airloop.keys.each do |info_key|
+              if info_key.include?('fan_power_credit')
+                if !user_airloop[info_key].to_s.empty?
+                  if info_key.include?('has_')
+                    if user_airloop[info_key].downcase == 'yes'
+                      air_loop.thermalZones.each do |thermal_zone|
+                        if thermal_zone.additionalProperties.hasFeature(info_key)
+                          current_value = thermal_zone.additionalProperties.getFeatureAsDouble(info_key).to_f
+                          thermal_zone.additionalProperties.setFeature(info_key, current_value + 1.0)
+                        else
+                          thermal_zone.additionalProperties.setFeature(info_key, 1.0)
+                        end
+                      end
+                    end
+                  else
+                    air_loop.thermalZones.each do |thermal_zones|
+                      if thermal_zone.additionalProperties.hasFeature(info_key)
+                        current_value = thermal_zone.additionalProperties.getFeatureAsDouble(info_key).to_f
+                        thermal_zone.additionalProperties.setFeature(info_key, current_value + user_airloop_doas[info_key])
```

Comment:
Should this be user_airloop[info_key]? The doas version does not seem to be defined here.